src/switchroot/ostree-prepare-root.c

```diff
@@ -151,12 +151,17 @@ main(int argc, char *argv[])
   if (chdir (deploy_path) < 0)
     err (EXIT_FAILURE, "failed to chdir to deploy_path");
 
-  /* In the systemd case, this is handled by ostree-system-generator */
+  bool mount_var = true;
+  /* In the systemd case, this is handled by ostree-system-generator by default */
 #ifndef HAVE_SYSTEMD_AND_LIBMOUNT
+  /* file in /run can override that behaviour */
+  if (lstat ("/run/ostree-mount-deployment-var", &stbuf) < 0)
+    mount_var = false;
```

Comment:
I also think we need to have this bit outside of the `#ifndef HAVE_SYSTEMD_AND_LIBMOUNT` right?  I'm not sure how this could have worked as we're building Fedora with that conditional set right?

Reply:
Ouch, yes, you're correct this is the exact opposite. I think the double negation with `#ifndef` confused me. Fixup coming up!

Reply:
Split this out into https://github.com/ostreedev/ostree/issues/1667